### PR TITLE
Update Java ExpansionService to use arbitrary PipelineOptions set through an ExpansionRequest

### PR DESCRIPTION
**Pull request by chamikaramj**
This fixes https://github.com/apache/beam/issues/33022.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by chamikaramj** [comment]
R: @ahmedabu98

File: sdks/java/expansion-service/src/test/java/org/apache/beam/sdk/expansion/service/ExpansionServiceTest.java [hunk 1]
```diff
@@ -146,6 +206,58 @@ public void testConstruct() {
...
+            .setComponents(pipelineProto.getComponents())
+            .setPipelineOptions(optionsStruct)
+            .setTransform(
+                RunnerApi.PTransform.newBuilder()
+                    .setUniqueName(TEST_NAME)
+                    .setSpec(RunnerApi.FunctionSpec.newBuilder().setUrn(TEST_OPTIONS_URN))
+                    .putInputs("input", inputPcollId))
+            .setNamespace(TEST_NAMESPACE)
+            .build();
+    ExpansionApi.ExpansionResponse response = expansionService.expand(request);
```
[the remark this answers is not in the document]
Comment (chamikaramj) [reply sdks/java/expansion-service/src/test/java/org/apache/beam/sdk/expansion/service/ExpansionServiceTest.java:250]: Yeah, the `expansionService.expand` call results in the transform provided in the `ExpansionRequest` being expanded.

**Review by ahmedabu98 — APPROVED** [review]
LGTM

**Comment by chamikaramj** [comment]
Thanks!

**Comment by chamikaramj** [comment]
Cdap failure in "Java IOs Direct" is unrelated.

**Comment by chamikaramj** [comment]
Java PipelineOptions deserialization expects the encoded options to have a specific key.

https://github.com/apache/beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/util/construction/PipelineOptionsTranslation.java#L99

```
"beam:option:" + snake_case_options_key + ":v1"
```

May be options forwarded from Python are not in this format ?

**Comment by ahmedabu98** [comment]
Python translation does put them in this format: https://github.com/apache/beam/blob/2488ca131bed3f28c92fa7b38b7506a461818a3a/sdks/python/apache_beam/options/pipeline_options.py#L427-L433

**Comment by ahmedabu98** [comment]
The issue might be that pipeline options are ignored when building the Java SchemaTransform: https://github.com/apache/beam/blob/2488ca131bed3f28c92fa7b38b7506a461818a3a/sdks/java/expansion-service/src/main/java/org/apache/beam/sdk/expansion/service/ExpansionServiceSchemaTransformProvider.java#L94C69-L94C76

**Comment by ahmedabu98** [comment]
Nevermind ignore my comments. The options are getting through to Java, but are somehow not getting picked up in step in the IO. I think the changes in this PR are working though

**Comment by ahmedabu98** [comment]
Looks like it's a general Java issue. Opened #33074

**Pull request merged**